planning/behavior_path_planner/autoware_behavior_path_planner_common/src/utils/path_safety_checker/safety_check.cpp

```diff
@@ -594,100 +720,13 @@ std::vector<Polygon2d> get_collided_polygons(
   std::vector<Polygon2d> collided_polygons{};
   collided_polygons.reserve(target_object_path.path.size());
   for (const auto & obj_pose_with_poly : target_object_path.path) {
-    const auto & current_time = obj_pose_with_poly.time;
-
-    // get object information at current time
-    const auto & obj_pose = obj_pose_with_poly.pose;
-    const auto & obj_polygon = obj_pose_with_poly.poly;
-    const auto object_velocity = obj_pose_with_poly.velocity;
-
-    // get ego information at current time
-    // Note: we can create these polygons in advance. However, it can decrease the readability and
-    // variability
-    const auto & ego_vehicle_info = vehicle_info;
-    const auto interpolated_data = get_interpolated_pose_with_velocity_and_polygon_stamped(
-      predicted_ego_path, current_time, ego_vehicle_info);
-    if (!interpolated_data) {
-      continue;
-    }
-    const auto & ego_pose = interpolated_data->pose;
-    const auto & ego_polygon = interpolated_data->poly;
-    const auto ego_velocity = std::min(interpolated_data->velocity, max_velocity_limit);
-
-    const double ego_yaw = tf2::getYaw(ego_pose.orientation);
-    const double object_yaw = tf2::getYaw(obj_pose.orientation);
-    const double yaw_difference = autoware_utils::normalize_radian(ego_yaw - object_yaw);
-    if (std::abs(yaw_difference) > yaw_difference_th) continue;
-
-    // check intersects
-    if (boost::geometry::intersects(ego_polygon, obj_polygon)) {
-      if (collided_polygons.empty()) {
-        debug.unsafe_reason = "overlap_polygon";
-        debug.expected_ego_pose = ego_pose;
-        debug.expected_obj_pose = obj_pose;
-        debug.extended_ego_polygon = ego_polygon;
-        debug.extended_obj_polygon = obj_polygon;
-      }
-      collided_polygons.push_back(obj_polygon);
-
-      continue;
-    }
+    CollisionCheckDebug * debug_ptr = collided_polygons.empty() ? &debug : nullptr;
```

Comment:
(optional) We will only have debug information for one calculation (the last one that started before a collision was found).
Is this the intended behavior ? How about storing each debug information in a vector instead ?

Reply:
Now I understand thank you.
Since this is not so obvious from the code, maybe some comment can be added ?